server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -141,6 +162,18 @@ public void afterRefresh(boolean didRefresh) {
         }
     }
 
+    String uploadSegmentInfosSnapshot(String latestSegmentsNFilename, SegmentInfos segmentInfosSnapshot) throws IOException {
+        long localCheckpoint = indexShard.getEngine().getProcessedLocalCheckpoint();
+        String commitGeneration = latestSegmentsNFilename.substring((IndexFileNames.SEGMENTS + "_").length());
```

Comment:
`IndexFileNames.parseGeneration(latestSegmentsNFilename)`?

Reply:
Used `SegmentInfos.generationFromSegmentsFileName` instead. `IndexFileNames.parseGeneration` only works for segment files where name starts with `_`